pkg/generators/openapi/openapi_generator.go

```diff
@@ -315,6 +328,10 @@ func (g *OpenAPIGenerator) generateResourcePaths(prefix string,
 func (g *OpenAPIGenerator) generateMethod(path []*concepts.Locator, method *concepts.Method) {
 	g.buffer.StartObject(strings.ToLower(g.binding.Method(method)))
 	g.generateDescription(method.Doc())
+	// Mark as deprecated if either the method or its parent resource is deprecated
+	if annotations.IsDeprecated(method) || annotations.IsDeprecated(method.Owner()) {
+		g.buffer.Field("deprecated", true)
+	}
```

Comment:
Nit: Any ways we can make this robust to several levels of nesting?

Reply:
What is the expected behavour you're looking for?

Let me give a concrete example:
```
/api/clusters_mgmt/v1/control_plane
```
Endpoint will soon have a GET and UPDATE to support Control Plane backup disable/enable.
Let's image some day this endpoint gets deprecated.
That does *NOT* mean that the `control_plane/upgrade_policies` is also deprecated!
i.e. sub resources do not automatically inharite deprecation annotation from their ancestor.